app/Http/Controllers/api/TagController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers\api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\TagRequest;
+use App\Http\Resources\TagResource;
+use App\Models\Tag;
+use Illuminate\Http\Exceptions\HttpResponseException;
+
+class TagController extends Controller
+{
+    public function index()
+    {
+        $tags = Tag::get();
+
+        if ($tags->isEmpty()) {
+            throw new HttpResponseException(response()->json(['message' => __('No hi ha tags a la base de dades')], 404));
+        }
+
+        return response()->json(['data' => TagResource::collection($tags)], 200);
+    }
+
+    public function store(TagRequest $request)
+    {
+        $tag = Tag::create($request->validated());
+
+        if ($tag) {
+            return response()->json(['data' => $tag, 'message' => __('Registre realitzat amb èxit.')], 201);
+        } else {
+            return response()->json(['message' => __('Error en crear el tag. Si-us-plau, torna-ho a provar.')], 500);
+        }
+    }
+
+    private function findTag($id)
+    {
+        $tag = Tag::findOrFail($id);
+
+        return $tag;
+    }
+
+    public function show($id)
+    {
+        $tag = $this->findTag($id);
```

Comment:
I think that the method `findTag`, with just one line, it's not necessary, at least at the moment. 
```suggestion
        $tag = Tag::findOrFail($id);
```

Reply:
I can't remember now why I wrote this code, probably it was mandatory for another thing, anyway i agree having extra codelines is not the best. I'll leave it like this at the moment. Tnxs!